src/main/java/com/minecolonies/core/research/LocalResearchTree.java

```diff
@@ -31,6 +38,11 @@
  */
 public class LocalResearchTree implements ILocalResearchTree
```

Comment:
Would be better to move the tracking/logic into a quest specific class, like a questmanager or the likes to avoid having quest code spread too much around outside of the quests package. Here we'd just have a "onresearchcompleted" callback to that questmanager which is doing the tracking